app/src/main/java/org/gdg/frisbee/android/arrow/ArrowTaggedActivity.java

```diff
@@ -76,51 +75,65 @@ public void onCreate(Bundle savedInstanceState) {
         taggedList.setLayoutManager(new LinearLayoutManager(this));
         taggedList.addItemDecoration(new DividerItemDecoration(this, null));
         taggedList.setAdapter(adapter);
+    }
 
-        AppStateManager.load(getGoogleApiClient(), Const.ARROW_DONE_STATE_KEY)
-                .setResultCallback(new ResultCallback<AppStateManager.StateResult>() {
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (item.getItemId() == android.R.id.home) {
+            finish();
+            return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    public void onConnected(Bundle bundle) {
+        super.onConnected(bundle);
+
+        if (TextUtils.isEmpty(serializedOrganizers)) {
+            loadFromSnapshot();
+        }
+    }
+
+    private void loadFromSnapshot() {
+        Games.Snapshots.open(getGoogleApiClient(), Const.GAMES_SNAPSHOT_ID, false).setResultCallback(
+                new ResultCallback<Snapshots.OpenSnapshotResult>() {
 
                     @Override
-                    public void onResult(AppStateManager.StateResult stateResult) {
-                        AppStateManager.StateConflictResult conflictResult = stateResult.getConflictResult();
-                        AppStateManager.StateLoadedResult loadedResult = stateResult.getLoadedResult();
-                        serializedOrganizers = "";
+                    public void onResult(Snapshots.OpenSnapshotResult stateResult) {
+                        if (!stateResult.getStatus().isSuccess()) {
+                            return;
+                        }
 
+                        Snapshot loadedResult = stateResult.getSnapshot();
+                        serializedOrganizers = "";
                         if (loadedResult != null) {
-                            final int statusCode = loadedResult.getStatus().getStatusCode();
-                            if (statusCode == AppStateStatusCodes.STATUS_OK) {
-                                serializedOrganizers = new String(loadedResult.getLocalData());
+                            try {
+                                serializedOrganizers = new String(loadedResult.getSnapshotContents().readFully());
+                            } catch (IOException e) {
+                                Timber.w(e, "Could not store tagged organizer");
+                                Toast.makeText(ArrowTaggedActivity.this, R.string.arrow_oops, Toast.LENGTH_LONG).show();
                             }
-                        } else if (conflictResult != null) {
```

Comment:
AppStateManager code is removed here. 
Conflicts are also removed. This is barely used and I think there is no need to show conflicted data here.